algorithms/HPdclassifier/R/predict.hpdegbm.R

```diff
@@ -0,0 +1,218 @@
+        library(gbm)
+
+        # Gaussian distribution for regression
+        if (appType == "regression") {
+           # Prediction by the first sub-model: AdaBoost, Bernoulli distribution for binary classification
+           GBM_modeli <- GBM_model[[1]]
+           best.iteri <- best.iter[1,1]
+           predi <- predict(GBM_modeli, newdata, best.iteri, type="link")  
+
+           # fusion with other sub-models
+           npartition_train <- nrow(best.iter)
+           for (k in 2: npartition_train) {
+               GBM_modelk <- GBM_model[[k]]
+               best.iterk <- best.iter[k,1]
+               predi <- predi + predict(GBM_modelk, newdata, best.iterk, type="link")
+           } 
+           predi <- predi/npartition_train # average of regressions from sub-models
+         }
+
+         # bernoulli or AdaBoost distribution for binary classification
+         if (appType == "binary-classification") {
+           # Prediction by the first sub-model: AdaBoost, Bernoulli distribution for binary classification
+           GBM_modeli <- GBM_model[[1]]
+           best.iteri <- best.iter[1,1]
+           predi < - sign(predict(GBM_modeli, data2, best.iteri, type)) # AdaBoost/Bernoulli 
+
+           # fusion with other sub-models
+           npartition_train <- nrow(best.iter)
+           for (k in 2: npartition_train) {
+               GBM_modelk <- GBM_model[[k]]
+               best.iterk <- best.iter[k,1]
+               predi <- predi + sign(predict(GBM_modelk, data2, best.iterk, type))
+           } 
+           predi <- sign(predi)
+         }
+
+
+         # multinomial distribution for multi-class classification
+         if (appType == "multi-classification") {
+           # Prediction by the first sub-model
+           # X: dframe/darray; Y: dframe (factor vector originally)/darray
+           GBM_modeli <- GBM_model[[1]]
+           best.iteri <- best.iter[1,1]
+           pred0 <- predict(GBM_modeli, data2, best.iteri, type="response") # fuzzy classification of multi-class
+
+           # fusion with other sub-models
+           npartition_train <- nrow(best.iter)
+           for (k in 2: npartition_train) {
+               GBM_modelk <- GBM_model[[k]]
+               best.iterk <- best.iter[k,1]
+               pred0 <- pred0 + (predict(GBM_modelk, data2, best.iterk, type="response")) # fusion of sub-models
+           } 
+           predi <- apply(pred0,1,which.max)
+         }
+
+        update(predi)
+     }) 
+
+     Predictions <- getpartition(daPredict)
+   } else{
+     # centralized prediction for small data (newdata). newdata is data.frame or matrix
+     nTest <- nrow(newdata)
+     GBM_model=model
+
+     # centralized prediction: newdata is a data.frame or matrix
+     library(gbm)
+
+     # Gaussian distribution for regression
+     if (appType == "regression") {
```

Comment:
it would be nice instead of repeating the same code so many times to simply create a function